DockDoor/Components/MouseTrackingView.swift

```diff
@@ -0,0 +1,34 @@
+import SwiftUI
+
+struct MouseTrackingView: NSViewRepresentable {
+    func makeNSView(context: Context) -> MouseTrackingNSView {
+        MouseTrackingNSView()
+    }
+
+    func updateNSView(_ nsView: MouseTrackingNSView, context: Context) {}
+}
+
+class MouseTrackingNSView: NSView {
+    override init(frame frameRect: NSRect) {
+        super.init(frame: frameRect)
+        setupTrackingArea()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupTrackingArea()
+    }
+
+    private func setupTrackingArea() {
+        let options: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways, .inVisibleRect]
+        let trackingArea = NSTrackingArea(rect: bounds, options: options, owner: self, userInfo: nil)
+        addTrackingArea(trackingArea)
+    }
+
+    override func mouseExited(with event: NSEvent) {
+        DispatchQueue.main.async {
+            SharedPreviewWindowCoordinator.shared.hideWindow()
```

Comment:
You missed the check
```
currentAppUnderMouse == nil || currentAppUnderMouse != lastAppUnderMouse
```
This means that if you stand on an app icon, go to a preview, and return to the icon, the preview will close instead of staying open.

Reply:
I found that it didn't require this check anymore, when I hover back on the icon it re-appears properly. Adding the check back is trivial, I will iron it out today.